app/controllers/taxon_frameworks_controller.rb

```diff
@@ -0,0 +1,81 @@
+class TaxonFrameworksController < ApplicationController
+  before_filter :authenticate_user!, :except => [:index, :show]
+  before_filter :curator_required, :only => [:new, :create, :edit, :update, :destroy]
+  before_filter :admin_required, :only => [:new, :create, :edit, :update, :destroy]
```

Comment:
If you have admin_required, you don't need curator_required, b/c if it requires the admin role, most curators won't be able to access it anyway.